promql/promqltest/test_test.go

```diff
@@ -558,7 +559,7 @@ eval range from 0 to 5m step 5m testmetric
 
 	for name, testCase := range testCases {
 		t.Run(name, func(t *testing.T) {
-			err := runTest(t, testCase.input, NewTestEngine(false, 0, DefaultMaxSamplesPerQuery))
+			err := runTest(&struct{ testutil.T }{t}, testCase.input, NewTestEngine(false, 0, DefaultMaxSamplesPerQuery))
```

Comment:
Would be good to add a comment here explaining why we're using `&struct{ ...` rather than just passing `t`.

Reply:
Is the point to defeat the type assertion I asked about?
This seems to have confused everybody that read it; why not pass a boolean saying whether to call `t.Run`, or remove the call from the line above and always do it?